tests/SigningTest.php

```diff
@@ -74,14 +73,20 @@ public function testSigv4HeaderSigning() {
             'credentials_provider' => $credentials_provider,
             'region' => self::SIGV4TEST_REGION,
             'service' => self::SIGV4TEST_SERVICE,
+            'date' => $date,
         ]);
         $http_request = new Request('GET', '/', [], ['Host' => 'example.amazonaws.com']);
         $this->assertNotNull($http_request, "Unable to create HttpRequest for signing");
         $signable = Signable::fromHttpRequest($http_request);
         $this->assertNotNull($signable, "Unable to create signable from HttpRequest");
 
-        Signing::signRequestAws($signable, $signing_config, function() {
-
+        Signing::signRequestAws($signable, $signing_config, function($signing_result, $error_code) use (&$http_request) {
+            $this->assertEquals(0, $error_code);
+            $signing_result->applyToHttpRequest($http_request);
         });
+
+        $headers = $http_request->headers();
+        $this->assertEquals('AWS4-HMAC-SHA256 Credential=AKIDEXAMPLE/20150830/us-east-1/service/aws4_request, SignedHeaders=host;x-amz-date, Signature=5fa00fa31553b73ebf1942676e86291e8372ff2a2260956d9b8aae1d763fbf31', $headers->get('Authorization'));
```

Comment:
same with above

Reply:
In order to shrink this, I would have to heredoc it, and then strip newlines from it. That seems a bit counter to simplicity, eh?